Workiva/over_react
# UIP-1637 First pass at strong mode compliance

**Ultimate problem:**
#14 - OverReact should be strong-mode compliant!

**How it was fixed:**
1. [Some changes were made to the react-dart wrapper](https://github.com/cleandart/react-dart/pull/106).
2. Lots of explicit type casting.
3. Resolving misc. Dart analyzer warnings, errors and hints.
4. For `_FluxComponentMixin`... I made it implement `BatchedRedraws`, but I'm not 100% confident in that solution.  Would be great to get @maxwellpeterson-wf, @evanweible-wf or @trentgrover-wf to weight in on the validity of that.
5. Turned on strong mode in `.analysis_options`.

**Testing suggestions:**
1. Verify that all tests pass
2. Verify that demo components behave as expected
3. Try it out on some of our more complex internal Workiva components to verify that regressions are not being introduced.

**Potential areas of regression:**
Pretty much everything 😁 
---

> **FYA:** @greglittlefield-wf @aaronlademann-wf @jacehensley-wf @clairesarsam-wf @joelleibow-wf @maxwellpeterson-wf, @evanweible-wf or @trentgrover-wf
